app/src/main/java/com/android/template/ui/RouterImpl.java

```diff
@@ -14,7 +14,8 @@
 
 public final class RouterImpl implements Router {
 
-    public static final int CONTAINER_ID = R.id.activity_main_fragment_container;
+    public static final int ACTIVITY_MAIN_CONTAINER_ID = R.id.activity_main_fragment_container;
```

Comment:
Aha, sada postaje jasnije. Ti ćeš "znati" iz kojeg screna pozivaš koje metode na routeru? To je oke, ali imaš temporal-coupling onda jer "ti nešto znaš implicitno" a što nije skroz vidljivo u kodu, što ti može uzrokovati probleme.

Reply:
Nisam imao ideje kako da to drukčije napravim s obzirom na to da imam 3 activityja i svaki ima fragment_container